assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,679 @@
+              $scope.options.pickedPushToImageStreamTag = $scope.outputImageStream.tag;
+            }
+            angular.forEach(projectImageStreams, function(imageStream, name) {
+              var tagList = [];
+              if (imageStream.status.tags) {
+                imageStream.status.tags.forEach(function(item){
+                  tagList.push(item["tag"]);
+                });
+              }
+              $scope.pushTo.imageStreams.push(name);
+              $scope.pushTo.tags[name] = tagList;
+            });
+
+            if (selectFirstOption) {
+              $scope.options.pickedPushToImageStream = $scope.pushTo.imageStreams[0];
+              $scope.clearSelectedOutputTag();
+              // If defined Image Stream is not present in the Image Streams of picked Namespace set tag to empty string,
+              // so the user has to pick from a existing Image Streams.
+            } else if (!$scope.pushTo.imageStreams.contains($scope.options.pickedPushToImageStream)) {
+              $scope.options.pickedPushToImageStreamTag = "";
+            }
+          } else if (projectName === $scope.outputImageStream.namespace) {
+            $scope.pushTo.imageStreams.push($scope.outputImageStream.imageStream);
+            $scope.options.pickedPushToImageStream = $scope.outputImageStream.imageStream;
+            $scope.options.pickedPushToImageStreamTag = $scope.outputImageStream.tag;
+          } else {
+            $scope.options.optionsModel.pickedPushToImageStream = "";
+            $scope.options.optionsModel.pickedPushToImageStreamTag = "";
+          }
+        });
+      }
+    }
+
+    $scope.clearSelectedOutputTag = function() {
+      var tags = $scope.pushTo.tags[$scope.options.pickedPushToImageStream];
+      if (tags) {
+        $scope.options.pickedPushToImageStreamTag = _.find(tags, function(tag) { return tag == "latest" }) || tags[0]
+      } else {
+        $scope.options.pickedPushToImageStreamTag =  "latest";
+      }
+    }
+
+    // Check if the namespace is available. If so add him to available namespaces and remove him from unavailable 
+    $scope.checkNamespaceAvailability = function(ns, type) {
+      DataService.get("projects", ns, {}, { errorNotification: false})
+      .then(function() {
+        $scope.availableProjects.push(ns);
+      }, function(result) {
+        if (result.status === 403) {
+          $scope.alerts["load-" + ns] = {
+            type: "error",
+            message: "Project '" + ns + "' is not available for your account.",
+            details: "Reason: " + $filter('getErrorDetails')(result)
+          };
+        } else {
+          $scope.alerts["load-" + ns] = {
+            type: "error",
+            message: "An error occurred loading the " + ns + " project.",
+            details: "Reason: " + $filter('getErrorDetails')(result)
+          };
+        }
+      });
+    }
+
+    $scope.save = function() {
```

Comment:
Nice progress on all of this :smile: 

I'm still concerned about the size of methods like `$scope.save` & the number of things happening, particularly for long term maintenance...